tests/Test.ADAL.NET.Integration/DeviceProfileTests.cs

```diff
@@ -88,5 +88,51 @@ public void Initialize()
             Assert.AreEqual("some-access-token", result.AccessToken);
         }
 
+        [TestMethod]
+        public async Task FullCoveragePositiveTest()
+        {
+
+            MockHttpMessageHandler mockMessageHandler = new MockHttpMessageHandler()
+            {
+                Method = HttpMethod.Get,
+                Url = "https://login.microsoftonline.com/home/oauth2/devicecode",
+                ResponseMessage = MockHelpers.CreateSuccessDeviceCodeResponseMessage()
+            };
+
+            HttpMessageHandlerFactory.AddMockHandler(mockMessageHandler);
+
+            mockMessageHandler = new MockHttpMessageHandler()
+            {
+                Method = HttpMethod.Post,
+                Url = "https://login.microsoftonline.com/home/oauth2/token",
+                ResponseMessage = MockHelpers.CreateFailureResponseMessage("{\"error\":\"authorization_pending\"," +
+                                                               "\"error_description\":\"AADSTS70016: Pending end-user authorization." +
+                                                               "\\r\\nTrace ID: f6c2c73f-a21d-474e-a71f-d8b121a58205\\r\\nCorrelation ID: " +
+                                                               "36fe3e82-442f-4418-b9f4-9f4b9295831d\\r\\nTimestamp: 2015-09-24 19:51:51Z\"," +
+                                                               "\"error_codes\":[70016],\"timestamp\":\"2015-09-24 19:51:51Z\",\"trace_id\":" +
+                                                               "\"f6c2c73f-a21d-474e-a71f-d8b121a58205\",\"correlation_id\":" +
+                                                               "\"36fe3e82-442f-4418-b9f4-9f4b9295831d\"}")
+            };
+
+            HttpMessageHandlerFactory.AddMockHandler(mockMessageHandler);
+            HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler()
+            {
+                Method = HttpMethod.Post,
+                Url = "https://login.microsoftonline.com/home/oauth2/token",
+                ResponseMessage =
+                    MockHelpers.CreateSuccessTokenResponseMessage(TestConstants.DefaultUniqueId,
+                        TestConstants.DefaultDisplayableId, TestConstants.DefaultResource)
+            });
+
+            TokenCache cache = new TokenCache();
+            AuthenticationContext ctx = new AuthenticationContext(TestConstants.DefaultAuthorityHomeTenant, cache);
+            DeviceCodeResult dcr = await ctx.AcquireDeviceCodeAsync("some-resource", "some-client");
+            Assert.IsNotNull(dcr);
```

Comment:
add Asserts check the values in DeviceCodeResult to be same as the ones configured in the mock